### ISSUE TITLE
textDocument/references does not return usages of base class method

### ISSUE THREAD

**Issue by tom-anders** [clangd/clangd#881]
```
struct A { virtual void foo() = 0; };
struct B : A { void foo() override; }; // textDocument/references returns a->foo() and b->foo()

void test(A* a, B* b) {
    a->foo(); 

    b->foo(); // textDocument/references only returns b->foo() here, but NOT a->foo()
}
```

`clangd --version`:
```
clangd version 14.0.0 (https://github.com/llvm/llvm-project a91cfd1990255aea545683083329c2ad3fee4518)
Features: linux
Platform: x86_64-unknown-linux-gnu
```

**Comment by kadircet** [comment]
There's definitely a discrepancy here, and I'd actually expect the first one to **not** report a reference to `a->foo()` either. Since you are clearly invoking references on the method of a derived class. @usx95 what is the reasoning/model here?

Also what is your expected behaviour in this case @tom-anders ?

**Comment by HighCommander4** [comment]
> There's definitely a discrepancy here, and I'd actually expect the first one to **not** report a reference to `a->foo()` either. Since you are clearly invoking references on the method of a derived class. @usx95 what is the reasoning/model here?

The discussion in #668 suggests that reporting the reference to `a->foo()` is the intended behaviour implemented in that issue.

To me, it makes sense: the call `a->foo()` _could_ call `B::foo()`, so it's a potential reference.

**Comment by tom-anders** [comment]
@kadircet I agree with @HighCommander4, IMO `a->foo()` should be reported in both cases.

**Comment by kadircet** [comment]
OK so we were discussing the behaviour with the team again. We all agree on showing the calls to any of the ancestors in the override chain, including the method itself, in the references call. We are a little bit divided on the declaration/definition front, but I think it is best to keep the current behaviour we have today, show declarations for all the ancestors and the first level of descendants, in addition to decl of the current method itself.

Anyway, this is clearly a bug that needs fixing in the direction you've pointed, refs on `b->foo()` should contain `a->foo()` as well.

**Comment by usx95** [comment]
Sent out https://reviews.llvm.org/D111039
